src/github.com/matrix-org/dendrite/syncapi/storage/syncserver.go

```diff
@@ -216,10 +216,10 @@ func (d *SyncServerDatabase) syncStreamPositionTx(
 }
 
 // IncrementalSync returns all the data needed in order to create an incremental
-// sync response for the given device. If the device has a deviceID events
-// returned will include any client transaction IDs associated with the device.
-// These transaction IDs come from when the device sent the event via an API
-// that included a transaction ID.
+// sync response for tshe given user. Events returned will include any client
+// transaction IDs associated with the given device. These transaction IDs come
```

Comment:
... but the device can be nil, in which case...

Reply:
I don't think the device can be nil here, since we use it to get at the `UserID` and it has come (almost) directly from the rest layer?

Reply:
Hmm, should we be passing it by pointer at this point?

Reply:
> Hmm, should we be passing it by pointer at this point?

well, that is the obvious question. I'd say not if it can't be nil.

Reply:
Yeah, fixed. Though in general I'm becoming cautious with how often we seem to be passing things by value as some of the structs seem quite big, but Device is tiny so shouldn't matter.